### PR TITLE
Fix registration access token cannot be deserialized

### PR DESCRIPTION
**Pull request by ovidiupopa91**
Closes gh-495

### CODE REVIEW

File: oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/authentication/OidcClientRegistrationAuthenticationProvider.java [hunk 1]
```diff
@@ -222,7 +223,8 @@ private OidcClientRegistrationAuthenticationToken registerClient(OidcClientRegis
 	private OAuth2Authorization registerAccessToken(RegisteredClient registeredClient, String issuer) {
 		JoseHeader headers = JwtUtils.headers().build();
 
-		Set<String> authorizedScopes = Collections.singleton(DEFAULT_CLIENT_CONFIGURATION_AUTHORIZED_SCOPE);
+		Set<String> authorizedScopes = Collections.unmodifiableSet(
```
Comment (jgrandja) [inline oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/authentication/OidcClientRegistrationAuthenticationProvider.java:226]: This could be changed to:

```java
		Set<String> authorizedScopes = Collections.unmodifiableSet(
				Collections.singleton(DEFAULT_CLIENT_CONFIGURATION_AUTHORIZED_SCOPE));
```

to reduce the extra `HashSet` creation but it may be confusing passing an immutable set to `Collections.unmodifiableSet()`. I think it would less confusing with:

```java
Set<String> authorizedScopes = new HashSet<>();
authorizedScopes.add(DEFAULT_CLIENT_CONFIGURATION_AUTHORIZED_SCOPE);
authorizedScopes = Collections.unmodifiableSet(authorizedScopes2);
```

What do you think?

Can you also please add a test. Thanks.

Comment (ovidiupopa91) [reply oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/authentication/OidcClientRegistrationAuthenticationProvider.java:226]: Sure @jgrandja , I'll do the change. 
I can change `authenticateWhenClientRegistrationRequestAndValidAccessTokenThenReturnClientRegistration` test to check if the authorized scope is unmodifiable. 

There is already an assertion checking that only one scope is available. 
```
assertThat(authorizationResult.getAccessToken().getToken().getScopes())
				.containsExactly("client.read");
```

I will assert that UnsupportedOperationException is thrown when a new element is added to the Set (it can be changed to `isUnmodifiable` from assertj when the version will be upgraded to `3.21.0`)

File: oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/oidc/authentication/OidcClientRegistrationAuthenticationProviderTests.java [hunk 2]
```diff
@@ -375,8 +375,9 @@ public void authenticateWhenClientRegistrationRequestAndValidAccessTokenThenRetu
 
 		// assert "registration" access token, which should be used for subsequent calls to client configuration endpoint
 		OAuth2Authorization authorizationResult = authorizationCaptor.getAllValues().get(0);
-		assertThat(authorizationResult.getAccessToken().getToken().getScopes())
-				.containsExactly("client.read");
+		Set<String> authorizedScopes = authorizationResult.getAccessToken().getToken().getScopes();
```
Comment (jgrandja) [inline oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/oidc/authentication/OidcClientRegistrationAuthenticationProviderTests.java:378]: Let's revert this change. Instead, configure the integration tests in `OidcClientRegistrationTests` to use `JdbcOAuth2AuthorizationService` instead of `InMemoryOAuth2AuthorizationService` and it will reproduce the error.

Comment (jgrandja) [reply oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/oidc/authentication/OidcClientRegistrationAuthenticationProviderTests.java:378]: @ovidiupopa91 Do you have time to apply this change by EOD tomorrow? I'd like to get this into this Wed's release. If you don't have time, I can apply it. Please let me know.

Comment (ovidiupopa91) [reply oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/oidc/authentication/OidcClientRegistrationAuthenticationProviderTests.java:378]: hi @jgrandja. I had some problems with my laptop. It's working now. I will apply the change right now

**Comment by jgrandja** [comment]
Thanks for the updates @ovidiupopa91 ! This is now in main.
